src/pgroonga-primary-maintainer.c

```diff
@@ -209,6 +213,83 @@ pgroonga_primary_maintainer_wal_size_check_all(void)
 	pgstat_report_activity(STATE_IDLE, NULL);
 }
 
+static bool
+checkPGroongaPrimaryMaintainerHours(char **newval,
+									void **extra,
+									GucSource source)
+{
+	char *rawstring;
+	List *elemlist;
+	ListCell *l;
+	int i = 0;
+	bool hours[HOURS_SIZE] = {};
+
+	for (i = 0; i < HOURS_SIZE; i++)
+	{
+		hours[i] = false;
+	}
+
+	if (*newval == NULL || *newval[0] == '\0')
+		return true;
+
+	rawstring = pstrdup(*newval);
+	if (!SplitDirectoriesString(rawstring, ',', &elemlist))
```

Comment:
Hmm...
Can we generalize `PGrnOptionParseNames()` and use it instead?
We don't want to allocate memory for this.